protocol/protos/NetRemoteService.proto

```diff
@@ -11,4 +11,7 @@ service NetRemote
     rpc WifiEnumerateAccessPoints (Microsoft.Net.Remote.Wifi.WifiEnumerateAccessPointsRequest) returns (Microsoft.Net.Remote.Wifi.WifiEnumerateAccessPointsResult);
     rpc WifiAccessPointEnable (Microsoft.Net.Remote.Wifi.WifiAccessPointEnableRequest) returns (Microsoft.Net.Remote.Wifi.WifiAccessPointEnableResult);
     rpc WifiAccessPointDisable (Microsoft.Net.Remote.Wifi.WifiAccessPointDisableRequest) returns (Microsoft.Net.Remote.Wifi.WifiAccessPointDisableResult);
+    rpc WifiAccessPointSetPhyType (Microsoft.Net.Remote.Wifi.WifiAccessPointSetPhyTypeRequest) returns (Microsoft.Net.Remote.Wifi.WifiAccessPointSetPhyTypeResult);
+    rpc WifiAccessPointSetAuthenticationMethod (Microsoft.Net.Remote.Wifi.WifiAccessPointSetAuthenticationMethodRequest) returns (Microsoft.Net.Remote.Wifi.WifiAccessPointSetAuthenticationMethodResult);
+    rpc WifiAccessPointSetEncryptionMethod (Microsoft.Net.Remote.Wifi.WifiAccessPointSetEncryptionMethodRequest) returns (Microsoft.Net.Remote.Wifi.WifiAccessPointSetEncryptionMethodResult);

```

Comment:
I'm wondering if we should consider consolidating these two. While each one has its own configuration value (in hostapd, and likely elsewhere), they are more or less linked. Since all fields are implicitly optionally in proto3, the client will have the option to specify either the authentication algorithm suite(s), the cipher suite(s), or both. Eg.

```proto
enum Dot11WepKeyType 
{
    WepKeyType40,
    WepKeyType140,
}
message Dot11WepKey
{
    Dot11WepKeyType Type;
    bytes Key;
}
message Dot11CipherSuiteConfiguration
{
    Dot11WepKey WepKey;
     // ... other cipher info for specific cipher algorithms     
}
message WifiAccessPointAuthenticationConfiguration
{
    // below map implies removing AuthenticationAlgorithm field from WifiAuthenticationMethodConfiguration message
    map<Dot11AuthenticationAlgorithm, WifiAuthenticationMethodConfiguration> Dot11AuthenticationAndKeyManagement AkmSuites;
    map<Dot11CipherAlgorithm, Dot11CipherSuiteConfiguration> CipherSuites;
}
```

Reply:
It makes to consolidate them. In Windows, there is a `DOT11_AUTH_CIPHER_PAIR` structure that does exactly that.

Not sure if I agree with using a map though since the API is for setting the new configuration for the AP, which would just be one key-value pair of "auth algo"-"auth config". Is there a reason to set multiple?

Reply:
I made some changes to combine the APIs but I did it slightly differently than what you proposed - please let me know what you think. I also didn't add WEP types yet since I'm not sure if the keys should be part of the auth method configuration or the encryption method configuration.

Reply:
> Not sure if I agree with using a map though since the API is for setting the new configuration for the AP, which would just be one key-value pair of "auth algo"-"auth config". Is there a reason to set multiple?

Multiple AKMs can be supported on the same radio (eg. WPA-Personal and WPA-Enterprise). That's why programs like hostapd accept multiple (eg. see the `wpa_key_mgmt` setting).

I'm ok with accepting/supporting only a single AKM at any one time, but the design shouldn't preclude being able to accept multiple simultaneously in the future.

Reply:
I ended up using `repeated` instead of `map`. It would make sense to have `map<Microsoft.Net.Wifi.Dot11AkmSuite, Microsoft.Net.Wifi.Dot11AkmSuiteConfiguration>`, but `Dot11AkmSuite` is an enum which can't be used as a map key and it seems silly to make a wrapper for that

Reply:
Good point about not be able to use an enum as a map key (the workaround is ugly).